sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TypeUtils.scala

```diff
@@ -42,13 +42,14 @@ object TypeUtils {
   }
 
   def checkForSameTypeInputExpr(types: Seq[DataType], caller: String): TypeCheckResult = {
-    if (types.distinct.size > 1) {
-      TypeCheckResult.TypeCheckFailure(
-        s"input to $caller should all be the same type, but it's " +
-          types.map(_.simpleString).mkString("[", ", ", "]"))
-    } else {
-      TypeCheckResult.TypeCheckSuccess
-    }
+    types.foreach ( t =>
+      if (!t.sameType(types.head)){
```

Comment:
(A space between `){`. This will not pass Scala style test.)

- from
```scala
if (!t.sameType(types.head)){
``` 

- to
```scala
if (!t.sameType(types.head)) {
```